doc/contribute/guidelines.rst

```diff
@@ -554,6 +570,38 @@ exceptions:
 * Avoid using binary literals (constants starting with ``0b``).
 * Avoid using non-ASCII symbols in code, unless it significantly improves
   clarity, avoid emojis in any case.
+* Use proper capitalization of nouns in code comments (e.g. ``UART`` and not
+  ``uart``, ``CMake`` and not ``cmake``).
+
+Beyond C code, the following coding style rules apply to other types of files:
+
+* CMake
+
+  * Indent with spaces, indentation is two spaces.
+  * No space between commands (e.g. ``if``) and the corresponding opening
+    bracket (e.g. ``(``).
+
+* Devicetree
+
+  * Indent with tabs.
+  * Follow the Devicetree specification conventions and rules.
+  * Use dashes (``-``) as word separators for node and property names.
+  * Use underscores (``_``) as word separators in node labels.
+  * Leave a single space on each side of the equal sign (``=``) in property
+    definitions.
+  * No empty lines before a dedenting ``};``.
+  * Single empty line separating nodes at the same hierarchy level.
+
```

Comment:
seemingly we need something to prevent random newline spacing in the same parts i.e. https://github.com/zephyrproject-rtos/zephyr/pull/83310/files#diff-a188c48a9493b631f516e6672a23b4792710f6c3e7375e7b4956957a9c22d3edR18

Reply:
I will leave this for a a second pass at it, don't want to extend the scope right now.

Reply:
This is a clear example of being too pick. Having a new line on that part makes easy to ready IMHO.
Why do we need this kind of rules? What we gain with that?

Reply:
> This is a clear example of being too pick. Having a new line on that part makes easy to ready IMHO. Why do we need this kind of rules? What we gain with that?

The files in that PR are a complete mess, compare that with other board files, it's completely at odds with that and doesn't even use the same layout in the same file, changing multiple times. It does not look easy to read

Reply:
From your pr:

> Because makes it more readable.

So to answer:

> Why do we need this kind of rules? What we gain with that?

We need it so people i.e. like you in your PR don't come up with your own styles completely at odds with what is standard in zephyr, we don't need every board or every subsystem to have their own coding styles which add nothing

Reply:
>Why do we need this kind of rules? What we gain with that?

Because we have had serious disagreements lately between contributors and maintainers. Having as much as possible documented saves us from having more debates around all these topics.

Reply:
First, I don't like the way you start to talk here. I think you are better than this.
Second, you can achieve what you are proposing without been too strict. Have the possibility to group things don't make code worst to read.
Third, nobody thinks the same way and you should respect other opinions, even more when you don't like it. 

I can be wrong, ignore if is that the case, but my impression is that you want to have some other tool than checkpatch. The problem is that if you start to restrict code freedom to achieve your goal that only shows that the tool that you are picking is not capable to do the job. This means that you are doing the tool job instead the contrary.

I have the opinion that people could propose to increase a little the bar on the field they are responsible. Otherwise, it is insane. Imagine that you start to use the same bar on areas that you are not a maintainer. Then the maintainer should respect your opinion on that? Forcing people to do what they don't want never brings good things.

This is why I believe Zephyr should have guidelines that are not too strict.

Reply:
There is no issues about having spaces inside a node. Even mainline Linux have it:

https://github.com/torvalds/linux/blob/fc033cf25e612e840e545f8d5ad2edd6ba613ed5/arch/arm/boot/dts/microchip/at91sam9n12ek.dts#L268-L283

https://github.com/torvalds/linux/blob/fc033cf25e612e840e545f8d5ad2edd6ba613ed5/arch/arm/boot/dts/microchip/sama5d2.dtsi#L44-L58

Reply:
I don't think the rules say anything about extra empty lines inside a node @nandojve. It only says that you cannot leave an empty line after a closing `};` and that you must leave an empty lines between nodes. But here you are showing empty lines **within** a node and between properties, which is not forbidden at all by the rules above, unless I am mistaken.

Reply:
Hi @carlescufi ,

I'm advocating against below idea from @nordicjm, beginning of this thread. Copying here to easy follow:

> seemingly we need something to prevent random newline spacing in the same parts i.e. https://github.com/zephyrproject-rtos/zephyr/pull/83310/files#diff-a188c48a9493b631f516e6672a23b4792710f6c3e7375e7b4956957a9c22d3edR18

He is using my PR as example of something that project should block. I'm not in favor of his idea and you can see that this pattern exists on mainline Linux too.

So, there is no good reason to be so strict. And this is what I've been saying all the time. IMHO this is not a problem and never will be a problem that we should complain on a review.

Reply:
We should have common styles for files and that looks like this https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/96boards/argonkey/96b_argonkey.dts
No random lines between aliases, no random lines between things inside the same node, **you are making up your own style for your PR which is not how other files are formatted**, something we absolutely should prevent

Reply:
@nordicjm ,

About the new topic that you brought: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/96boards/argonkey/96b_argonkey.dts#L104

You are 100% right about this. I totally agree with you about it. This is not a good way to define a node in devicetree and should be fixed.

> you are making up your own style for your PR which is not how other files are formatted, 

> And to flip the question around, why is it so important/required that your dts files differ from the norm. in zephyr which has been fine for years?

With all respect, I can say, in fact, this is you. We have been using this for many years on Zephyr. The way you are writing you are distorting the facts. This is not different of the norm. I gave an example that this is not only used here but outside, inclusive, and that is a project of some relevance for devicetree.

I think you are simple ignoring the facts to try push your vision. If it is not new, how can I inventing or even pushing something different from the norm?

>  something we absolutely should prevent.

With this you are creating your own style, not me. You want to block something that has been used here and outside for a long time. Why?

I think you are the guy that is asking too much here. I'm saying that we should not touch this part because it is a common practice for many years here and on Linux and don't affect visually the node. The second case that you brought is different scenario and I agreed with you in that case.

Reply:
Can either of you provide some stats perhaps? if most of the current code complies with one approach or other perhaps we can include it in the list. Otherwise I would rather not tread there to avoid overloading this PR.

Reply:
As I understand it, @nandojve does not want us to introduce guidelines that discourage empty lines between properties within a node. 

I agree. Grouping properties using newlines is common practice in DTS, both in Zephyr and elsewhere. 

I don't currently see anything in the content of this PR for introducing such guidelines, so perhaps we can leave it out of this PR - and address it separately, if anybody desires?

Reply:
> Can either of you provide some stats perhaps? if most of the current code complies with one approach or other perhaps we can include it in the list. Otherwise I would rather not tread there to avoid overloading this PR.

Out of 886 board dts files in zephyr, the following have spaces:
```
boards/mediatek/mt8196/mt8196_adsp.dts line 107
boards/microchip/mec1501modular_assy6885/mec1501modular_assy6885.dts line 25
boards/mxchip/az3166_iotdevkit/az3166_iotdevkit.dts line 19
boards/mxchip/az3166_iotdevkit/az3166_iotdevkit.dts line 23
boards/others/serpente/serpente.dts line 30
```
4 out of 886 = 0.4%

Reply:
Can you put your script here? It seems to be too selective.

https://github.com/zephyrproject-rtos/zephyr/blob/dabeb9c95c719eb325b911c0e325d52b4a9dc244/boards/atmel/sam/sam4l_ek/sam4l_ek.dts#L52-L58

https://github.com/zephyrproject-rtos/zephyr/blob/dabeb9c95c719eb325b911c0e325d52b4a9dc244/boards/atmel/sam/sam_e70_xplained/sam_e70_xplained-common.dtsi#L57-L62

Is this change only valid for the boards directory? Or we are looking only there because @nordicjm is a collaborator of that part.

BTW who is 57300 ?
https://github.com/zephyrproject-rtos/zephyr/blob/dabeb9c95c719eb325b911c0e325d52b4a9dc244/MAINTAINERS.yml#L594-L602

Reply:
As said, only for aliases, https://pastebin.com/tuv8VLsK

> Is this change only valid for the boards directory? Or we are looking only there because @nordicjm is a collaborator of that part.

It's there because that's where board aliases are